lib/driver.js

```diff
@@ -332,6 +324,14 @@ class IosDriver extends BaseDriver {
       return;
     }
 
+      // if we have an ipa file, set it in opts
+      if (this.opts.app) {
```

Comment:
Yes, this looks good to me. There is no reason to do the ipa check earlier. 

As for conceptual debt, I agree. This was largely copied over from the old code. It might be good to encapsulate this logic into an `App` class or something. But it is more of a change than a bugfix, and given our current structure, this patch is good.